src/scene/immediate/immediate.js

```diff
@@ -234,57 +229,47 @@ class Immediate {
     }
 
     getGraphNode(matrix) {
-        let graphNode = null;
-        if (this.freeGraphNodes.length > 0) {
-            graphNode = this.freeGraphNodes.pop();
-        } else {
-            graphNode = new GraphNode();
-        }
-        this.usedGraphNodes.push(graphNode);
-
+        const graphNode = new GraphNode();
         graphNode.worldTransform = matrix;
         graphNode._dirtyWorld = graphNode._dirtyNormal = false;
 
         return graphNode;
     }
 
-    // called before the frame is rendered, prepares data for rendering
-    onPreRender() {
+    // called before the layer is getting rendered, this updates debug rendering for the layer
+    // This is called just before the layer is rendered to allow lines to be added from inside
+    // the frame getting rendered
```

Comment:
This comment probably needs to be tightened up a bit. Says the same thing twice.

Reply:
done.